libqpdf/QPDF_Array.cc

```diff
@@ -46,6 +46,96 @@ Array::array() const
     return nullptr; // unreachable
 }
 
+Array::iterator
```

Comment:
This is a lot better than aItems and dItems (vs. aitems and ditems). It creates a good, idiomatic approach that can coexist with old code. I think we should write something in the docs about switching to the new style. This doesn't have to block release of 12.0, but we should get it in at some point. Maybe I should write it to test my understanding, and you can review it.

Reply:
At the moment these classes are qpdf private API. Before exposing it I want to fully implement the Array and Dictionary in case I find I have to tweak the API. At that time we will need to update the documentation as well as the example programs.

The general intention is that method comments in the header files should be written as methods are added / modified. I skipped this given the 12.0 target date but I'll need to catch up once this is out of the way.

Reply:
Okay, we can work these in gradually throughout 12.x as they will be additions rather than ABI-breaking changes. I guess that's what you meant in your summary.